Microsoft.Toolkit.Uwp.UI.Controls/InAppNotification/InAppNotification.cs

```diff
@@ -74,15 +74,21 @@ protected override void OnApplyTemplate()
         /// <param name="duration">Displayed duration of the notification in ms (less or equal 0 means infinite duration)</param>
         public void Show(int duration = 0)
         {
-            _timer.Stop();
+            _animationTimer.Stop();
+            _dismissTimer.Stop();
 
             Visibility = Visibility.Visible;
             VisualStateManager.GoToState(this, StateContentVisible, true);
+            Opening?.Invoke(this, EventArgs.Empty);
```

Comment:
Can we put this before the visibility change and check the eventargs.handled?  This way if they register to the event they could prevent the popup from displaying.

Reply:
@michael-hawker I am not sure how to check the Handled property.

Reply:
You need to create your own EventArgs class which has a 'Handled' property on it.  Then, construct it and pass its reference into the invoke to Opening.  Then you can see if the property was set to true afterwards and escape.  

Though if we're going to align more with FlyoutBase in the future, we may want to use 'Cancel' as they do here for `Closing`: https://docs.microsoft.com/en-us/uwp/api/windows.ui.xaml.controls.primitives.flyoutbaseclosingeventargs

Reply:
I can directly start with `Cancel` right now, don't you think?

Reply:
Yes, sorry, that's kinda what I meant. :P